app/src/sharedTest/java/org/oppia/app/player/state/StateSelectionInteractionTest.kt

```diff
@@ -0,0 +1,150 @@
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import junit.framework.Assert.assertTrue
+import kotlinx.coroutines.CoroutineDispatcher
+import org.hamcrest.EasyMock2Matchers.equalTo
+import org.hamcrest.Matcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.home.HomeActivity
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPosition
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for SelectionInteraction [MULTIPLE_CHOICE_INPUT], [ITEM_SELECT_INPUT]. */
+@RunWith(AndroidJUnit4::class)
+class StateSelectionInteractionTest {
+
+  private val maxSelectionAllowedCount = 2
+  private var counter = 0
+  private lateinit var launchedActivity: Activity
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<HomeActivity> = ActivityTestRule(
+    HomeActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    val intent = Intent(Intent.ACTION_PICK)
+    launchedActivity = activityTestRule.launchActivity(intent)
+  }
+
+  @Test
+  fun testMultipleChoiceInput_showsRadioButtons_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOption() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        RecyclerViewActions.actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        RecyclerViewActions.actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(1, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOptions() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button1)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        RecyclerViewActions.actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        RecyclerViewActions.actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(2, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_withMaxSelectionAllowed_userSelectsDesiredOptions() {
```

Comment:
This test case is failing.

Reply:
It is passing I tested it. Please take a look once again.

Reply:
This test case is still failing

<img width="1456" alt="Screenshot 2019-10-29 at 9 02 39 PM" src="https://user-images.githubusercontent.com/9396084/67783807-ce024280-fa90-11e9-9ae1-73fb22ade6b1.png">


Error is mainly because the item that you are clicking is not visible on screen. First you will have to make it visible and then click on on that item.

Reply:
When I am running it. Its passing. I will check.

Reply:
This is failing while i checked

Reply:
This is handled in duplicate PR #276 .